Add BSSID connection options to PicoW

earlephilhower opened the pull request:
When working with a mesh or multiple APs with the same SSID, it is often necessary to specify which of the APs to connect to in order to maximize the WiFi strength.

Add BSSID options to the SDK's PicoW cyw43_arch_wifi_connect_XXX APIs.

Fixes #1090

peterharperuk commented:
Commit is missing reference to https://github.com/raspberrypi/pico-sdk/issues/1090
This is hitting a file that's targeted by a few PR's. I might push my own rebased version if that's ok.

peterharperuk commented:
Odd, doesn't work for me...

earlephilhower commented:
Sorry, @peterharperuk .  I think my fork's `develop` branch got into a weird state.

I have rebased manually to the tip of `develop` and added the `Fixes #xxx` comment to the git commit message.  Hope this helps.

earlephilhower commented:
Very odd.  I'll take a look at it tonight my time (~12hrs) and report back.  I believe I was able to get it to sync to a specified BSSID when I tried it last year, but will verify again and if it's busted will move this to a WIP PR so it doesn't pollute things.

peterharperuk commented:
Ok, my fixups are in here https://github.com/peterharperuk/pico-sdk/commits/bssid
I needed to apply this fix https://github.com/georgerobotics/cyw43-driver/pull/56
I used this to test it https://github.com/georgerobotics/cyw43-driver/pull/57

earlephilhower commented:
I spent some time this evening trying to get the latest develop + cyw43 driver + patch + the Arduino core running but ended up using your fork/branch directly .  (Seems like lots of things have advanced...)

Anyway, FWIW, I am not able to connect to any of my APs (not in mesh mode, but all with same SSID and different BSSIDs) when specifying the BSSID (even reversed in case there was some endianness-thing).  W/o BSSID all still seems okay under quick tests.

So I think for now, since you're not seeing the driver connecting to the specified BSSID and I myself can't seem to get any kind of connection w/BSSID, I will close this PR but leave the issue.  Sorry for the wild goose chase, and thanks for your effort.

peterharperuk commented:
ok, I got it working with a change to the driver, so I'll keep pushing to get the change in.

earlephilhower commented:
Much appreciate the effort, @peterharperuk!   This is a very useful feature in many instances.
